modules/nixos/macos.nix

```diff
@@ -49,4 +50,35 @@
     programs.htop.enable = true;
     programs.htop.settings.show_program_path = true;
   };
+
+  system.activationScripts.postActivation.text = let
+    authorizedKeysDeployUser = "/etc/ssh/authorized_keys.d/${config.deployUser}";
+    sshdAuthorizedKeysConf = "/etc/ssh/sshd_config.d/200-authorized-keys.conf";
+  in ''
+    mkdir -p $(dirname ${sshdAuthorizedKeysConf})
+    echo "AuthorizedKeysFile .ssh/authorized_keys /etc/ssh/authorized_keys.d/%u" > ${sshdAuthorizedKeysConf}
+    chmod 444 ${sshdAuthorizedKeysConf}
+
+    mkdir -p $(dirname ${authorizedKeysDeployUser})
+    echo > ${authorizedKeysDeployUser}
```

Comment:
Could just `touch` here?

Reply:
that wouldn't empty the file. my goal was to make this idempotent as is usually the case with nix based configuration.